core/src/main/kotlin/model/defaultValues.kt

```diff
@@ -3,9 +3,15 @@ package org.jetbrains.dokka.model
 import org.jetbrains.dokka.model.properties.ExtraProperty
 import org.jetbrains.dokka.model.properties.MergeStrategy
 
-class DefaultValue(val value: Expression): ExtraProperty<Documentable> {
+class DefaultValue(val expression: SourceSetDependent<Expression>): ExtraProperty<Documentable> {
+
+    @Deprecated("Use `expression` property that depends on source set", ReplaceWith("expression.values.first()"))
```

Comment:
Can you actually try out the replacement suggestion in IDEA? I think it should be with `this`, otherwise if you apply the fix the statement won't compile

There was such a similar problem with replacement for `documentable` vs `documentables`, so maybe it's the same here
```suggestion
    @Deprecated("Use `expression` property that depends on source set", ReplaceWith("this.expression.values.first()"))
```